### PR TITLE
[MM-16799] Introduce tests for the webapp side of the plugin

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR introduces tests for the webapp side of the plugin.

Field validation has changed to be more function-based, rather than relying on refs.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-16799

### CODE REVIEW

File: webapp/.babelrc [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (crspeller) [inline webapp/.babelrc]: can we do what I did with babel.config.js in the webapp instead of this?

File: webapp/package.json [hunk 2]
```diff
@@ -22,10 +24,17 @@
...
     "babel-eslint": "10.0.2",
     "babel-loader": "8.0.6",
+    "enzyme": "^3.10.0",
+    "enzyme-adapter-react-16": "^1.14.0",
+    "enzyme-to-json": "^3.3.5",
     "eslint": "5.16.0",
     "eslint-import-resolver-webpack": "0.11.1",
     "eslint-plugin-import": "2.18.0",
     "eslint-plugin-react": "7.14.2",
+    "jest": "^24.8.0",
```
Comment (crspeller) [inline webapp/package.json]: Exact versions please.

File: webapp/tests/helpers/client-test-helper.jsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+// See LICENSE.txt for license information.
+
+import jqd from 'jquery-deferred';
+import {Client4} from 'mattermost-redux/client';
+
+import WebSocketClient from 'client/websocket_client.jsx';
+
+var HEADER_TOKEN = 'token';
+
+class TestHelperClass {
```
Comment (crspeller) [inline webapp/tests/helpers/client-test-helper.jsx]: I don't know about copying in all these helpers. I feel like we should only copy the ones we need, or we should import them.

File: webapp/tests/setup.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import Adapter from 'enzyme-adapter-react-16';
+import {configure} from 'enzyme';
+import $ from 'jquery';
```
Comment (crspeller) [inline webapp/tests/setup.js]: I assume this is copy from webapp. But is this necessary?

**Comment by mickmister** [comment]
Build error is isolated to circleci. Lint and tests pass locally

**Pull request merged**